### ISSUE TITLE
django.middleware.locale.LocaleMiddleware required for cms_toolbar

### ISSUE THREAD

**Issue by marijngoedegebure** [divio/django-cms#4771]
From CMS version 3.1 two functions have been added to toolbar.py. These functions insert a bug for when you render the cms toolbar in a single language project.
When rendering django cms_toolbar it requires that the LANGUAGE_CODE is in the request, like this:

```
def get_object_draft_url(self):
        if self.obj:
            with force_language(self.request.LANGUAGE_CODE):
```

But as is stated in the code at the top of toolbar.py, whether language is in the request depends on whether multiple languages are enabled:

```
if settings.USE_I18N:
    self.language = get_language_from_request(request)
else:
    self.language = settings.LANGUAGE_CODE
```

I propose to adapt this same technique for both get_object_draft_url and get_object_public_url and let them rely on the language attribute of the toolbar.

**Comment by marijngoedegebure** [comment]
I created a pull request for this issue:

https://github.com/divio/django-cms/pull/4772

**Comment by yakky** [comment]
Fixed in #4950
